classes/ZipAction.php

```diff
@@ -175,48 +122,31 @@ private function compressWithZipArchive(&$filesList, $toFile)
         return true;
     }
 
-    private function compressWithPclZip(&$filesList, $toFile)
+    /**
+     * Extract an archive to the given directory
+     *
+     * @return bool success
+     *              we need a copy of it to be able to restore without keeping Tools and Autoload stuff
+     */
+    public function extract($from_file, $to_dir)
     {
-        $zip = $this->openWithPclZip($toFile);
-        if (!$zip) {
+        if (!is_file($from_file)) {
+            $this->logger->error($this->translator->trans('%s is not a file', array($from_file), 'Modules.Autoupgrade.Admin'));
+
             return false;
         }
 
-        $files_to_add = array();
-        for ($i = 0; $i < $this->configMaxNbFilesCompressedInARow && count($filesList); ++$i) {
-            $file = array_shift($filesList);
+        if (!file_exists($to_dir)) {
+            // ToDo: Use Filesystem from Symfony
+            if (!mkdir($to_dir)) {
+                $this->logger->error($this->translator->trans('Unable to create directory %s.', array($to_dir), 'Modules.Autoupgrade.Admin'));
 
-            $archiveFilename = $this->getFilepathInArchive($file);
-            if (!$this->isFileWithinFileSizeLimit($file)) {
-                continue;
+                return false;
             }
-
-            $files_to_add[] = $file;
-            $this->logger->debug($this->translator->trans(
-                '%filename% added to archive. %filescount% files left.',
-                array(
-                    '%filename%' => $archiveFilename,
-                    '%filescount%' => count($filesList),
-                ),
-                'Modules.Autoupgrade.Admin'
-            ));
-        }
-
-        $added_to_zip = $zip->add($files_to_add, PCLZIP_OPT_REMOVE_PATH, $this->prodRootDir);
-        $zip->privCloseFd();
-        if (!$added_to_zip) {
-            (new Filesystem())->remove($toFile);
-            $this->logger->error($this->translator->trans('[ERROR] Error on backup using PclZip: %s.', array($zip->errorInfo(true)), 'Modules.Autoupgrade.Admin'));
-
-            return false;
+            chmod($to_dir, 0775);
```

Comment:
Why forcing 775? 🤔

Reply:
As I'm not the original author of this line, I guess this is for restricting access to all users (mkdir is 777 and there is no umask in the project).

Reply:
I think you can remove all chmod like in this module since we remove them in the Core. Let apache / nginx user define rules :)

Reply:
Okay, but in a dedicated PR to let the QA make a proper review. I found other occurrences of `chmod` in the code.

Reply:
Created issue #291